src/vm/docs/mod.rs

```diff
@@ -358,6 +358,35 @@ value return by the successive applications.",
 (fold * (list 2 2 2) 0) ;; Returns 0"
 };
 
+const CONCAT_API: SpecialAPI = SpecialAPI {
+    input_type: "(buff, buff)|(list, list)",
+    output_type: "buff|list",
+    signature: "(concat buff-a buff-b)",
+    description: "The `concat` function takes two buffers or two lists with the same entry type, 
+and returns a concatenated buffer or list of the same entry type, with max_len = max_len_a + max_len_b.",
+    example: "(concat \"hello \" \"world\") ;; Returns \"hello world\""
+};
+
+const APPEND_API: SpecialAPI = SpecialAPI {
+    input_type: "list A, A",
+    output_type: "list",
+    signature: "(append (list 1 2 3 4) 5)",
+    description: "The `append` function takes a list and another value with the same entry type, 
+or a buffer and another buffer of length 1 and outputs a buffer or a list of the same type with max_len += 1.",
+    example: "(append (list 1 2 3 4) 5) ;; Returns (list 1 2 3 4 5)"
+};
+
+const ASSERTS_MAX_LEN_API: SpecialAPI = SpecialAPI {
+    input_type: "Function(A, B) -> B, (list A)",
+    output_type: "B",
+    signature: "(asserts-max-len buffer 10)",
+    description: "The `asserts-max-len` function takes a buffer or list argument, which must be typed as a list 
+or buffer of length N and outputs that same list or buffer, but typed with max length number-literal. 
```

Comment:
I would add to this description somewhere that the `length` argument must be a literal value.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1139/commits/d8eab4a97ca807bd8a4279635fd12321473d5d52